### PR TITLE
Added support for Time variable and dependency

### LINKED ISSUE #71: Get system time/date procedure blocks
**Describe the feature**

A blocks for procedures which would get number of year, month, day, hour, minute, (maybe second) (seperated into multiple blocks)

**How would you implement it?**

I would put a blocks into math category or advanced or something like that. I know it's possible to make this since minecraft is already using it for halloween and christmas events (pumpkin monsters, christmas chests) and java supports it (I tryed it today in eclipse with some tutorial and it worked)
**Additional context**

This would open abillities to make limited time mod features, like gear you can only get through certain period of time.

### PR DESCRIPTION
**Pull request by Goldorion**
This PR adds a new variable and dependency type, Time. This new variable type also adds some new getter to be able to use methods of the Date class (it adds my Date plugin with the official way). As this new variable is not a big one and doesn't have a reason to be used as a Global variable, I only added it for local variables. Also, if it's added to GV, I'm not sure if it will have good support.

Note: Procedure blocks are implemented in JSON as it's much more simple to make, but they use an input value to get a Time variable (with a normal getter or a getter from another procedure (maybe other PBs in the future)). Basic procedure blocks like the variable getter, setter and return value are, however, implemented in the code like the other ones.

Closes #71

### CODE REVIEW

File: plugins/generator-1.15.2/forge-1.15.2/procedures/_set_variable.java.ftl [hunk 1]
```diff
@@ -98,5 +98,7 @@
         ${name} = ${mappedMCItemToItemStackCode(value, 1)};
     <#elseif type == "BLOCKSTATE">
         ${name} = ${mappedBlockToBlockStateCode(value)};
+    <#elseif type == "TIME">
+        ${name} =(DATE)${value};
```
Comment (NOYBss) [inline plugins/generator-1.15.2/forge-1.15.2/procedures/_set_variable.java.ftl:102]:
```suggestion
        ${name} =(Date)${value};
```

**Review by NOYBss — CHANGES_REQUESTED** [review]
Typo 😄

**Comment by NOYBss** [comment]
Btw do you really need to cast to Date in _set_variable procedure block?

File: src/main/resources/blockly/js/mcreator_blocks.js [hunk 2]
```diff
@@ -111,6 +111,16 @@ Blockly.Blocks['procedure_retval_blockstate'] = {
     }
 };
 
+Blockly.Blocks['procedure_retval_time'] = {
+    init: function () {
+        this.appendDummyInput()
+            .appendField(javabridge.t("blockly.block.procedure_retval"))
+            .appendField(new Blockly.FieldDropdown(arrayToBlocklyDropDownArray(javabridge.getListOf("procedure_retval_time"))), 'procedure');
+        this.setOutput(true, 'MCItemBlock');
```
Comment (AlsoSomeoneElse) [inline src/main/resources/blockly/js/mcreator_blocks.js:119]: This should be 'Time'

**Review by AlsoSomeoneElse — CHANGES_REQUESTED** [review]

**Comment by Goldorion** [comment]
> 
> 
> Btw do you really need to cast to Date in _set_variable procedure block?

For most of the cases, surely not, but it can prevent some build errors, so we can keep it. It's not like if it changes something.

**Review by Max094Reikeb — APPROVED** [review]
Sounds good. Does this PR close #71 ? Because if it does, you should add the close keyword.

**Comment by Goldorion** [comment]
> 
> 
> Sounds good. Does this PR close #71 ? Because if it does, you should add the close keyword.

Oh yes, it closes the issue. I forgot there was an issue for that.

**Comment by Max094Reikeb** [comment]
Why did you merge it already? I'm the only one who reviewed, you should have wait for a second review...

**Comment by Goldorion** [comment]
> 
> 
> Why did you merge it already? I'm the only one who reviewed, you should have wait for a second review...

NOYB and SomeoneElse reviewed the PR and everything they asked have been fixed, so I thought it was ready to be merged.

**Comment by Max094Reikeb** [comment]
Anyways, it's too late now. If @NOYBss or @AlsoSomeoneElse see any problems, they can always do a fixing PR or report a bug.

**Comment by Goldorion** [comment]
> 
> 
> Anyways, it's too late now. If @NOYBss or @AlsoSomeoneElse see any problems, they can always do a fixing PR or report a bug.

Yeah and the PR doesn't change something, so it can't break workspaces.

**Pull request merged**